src/event/workqueue.c

```diff
@@ -293,6 +293,52 @@ _dispatch_workq_count_runnable_workers(dispatch_workq_monitor_t mon)
 
 	_dispatch_unfair_lock_unlock(&mon->registered_tid_lock);
 }
+#elif defined(__FreeBSD__)
+#include <sys/sysctl.h>
+#include <sys/proc.h>
+#include <sys/user.h>
+
+static void
+_dispatch_workq_count_runnable_workers(dispatch_workq_monitor_t mon)
+{
+	struct kinfo_proc kp[WORKQ_MAX_TRACKED_TIDS];
+	size_t size;
+	int count, runners = 0;
+	int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PID | KERN_PROC_INC_THREAD, (int)getpid()};
+
+	// get size we need
+	if (sysctl(mib, 4, NULL, &size, NULL, 0) < 0) {
+		_dispatch_debug("workq: failed to get size for kinfo_proc[] from sysctll");
+		return;
+	}
+
+	// only care about up to WORKQ_MAX_TRACKED_TIDS threads
+	size = size > sizeof(kp) ? sizeof(kp) : size;
```

Comment:
Why not write this using `MIN`?

```suggestion
	size = MIN(sizeof(kp), size);
```